src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -9,35 +11,125 @@
 import bio.terra.model.SnapshotBuilderCountResponse;
 import bio.terra.model.SnapshotBuilderCountResponseResult;
 import bio.terra.model.SnapshotBuilderGetConceptsResponse;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.ingest.IngestUtils;
+import bio.terra.service.filedata.azure.AzureSynapsePdao;
+import bio.terra.service.snapshotbuilder.utils.AggregateBQQueryResultsUtils;
+import bio.terra.service.snapshotbuilder.utils.AggregateSynapseQueryResultsUtils;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryDatasetPdao;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryPdao;
+import com.google.cloud.bigquery.TableResult;
+import java.sql.ResultSet;
 import java.util.List;
 import java.util.UUID;
+import java.util.function.Function;
+import org.apache.commons.lang3.NotImplementedException;
 import org.springframework.stereotype.Component;
 
 @Component
 public class SnapshotBuilderService {
 
   private final SnapshotRequestDao snapshotRequestDao;
+  private final DatasetService datasetService;
+  private final BigQueryDatasetPdao bigQueryDatasetPdao;
 
-  public SnapshotBuilderService(SnapshotRequestDao snapshotRequestDao) {
+  private final AzureSynapsePdao azureSynapsePdao;
+
+  public SnapshotBuilderService(
+      SnapshotRequestDao snapshotRequestDao,
+      DatasetService datasetService,
+      BigQueryDatasetPdao bigQueryDatasetPdao,
+      AzureSynapsePdao azureSynapsePdao) {
     this.snapshotRequestDao = snapshotRequestDao;
+    this.datasetService = datasetService;
+    this.bigQueryDatasetPdao = bigQueryDatasetPdao;
+    this.azureSynapsePdao = azureSynapsePdao;
   }
 
   public SnapshotAccessRequestResponse createSnapshotRequest(
       UUID id, SnapshotAccessRequest snapshotAccessRequest, String email) {
     return snapshotRequestDao.create(id, snapshotAccessRequest, email);
   }
 
-  public SnapshotBuilderGetConceptsResponse getConceptChildren(UUID datasetId, Integer conceptId) {
+  // TODO - replace manual query with query builder
+  // The query builder will handle switching between cloud platforms, so we won't
+  // have this duplicated logic in the final solution.
+  private String buildConceptChildrenQuery(
+      Dataset dataset, int conceptId, AuthenticatedUserRequest userRequest) {
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+    if (cloudPlatformWrapper.isGcp()) {
+      var bqTablePrefix =
+          String.format(
+              "%s.%s",
+              dataset.getProjectResource().getGoogleProjectId(),
+              BigQueryPdao.prefixName(dataset.getName()));
+
+      return String.format(
+          """
+                  SELECT c.concept_name, c.concept_id FROM `%s.concept` AS c%n\
```

Comment:
Spotbugs complained without it - `SnapshotBuilderService.java:68:68 FS: Format string should use %n rather than \n in ..buildConceptChildrenQuery(Dataset, int, AuthenticatedUserRequest) (M)`